useful_types/__init__.py

```diff
@@ -321,3 +321,23 @@ def __getitem__(self, __i: int) -> int:
 
 class SizedBuffer(Sized, Buffer, Protocol):
     ...
+
+
+# Source from https://github.com/python/typing/issues/256
```

Comment:
That thread is very long, maybe link to a specific message?

Other feedback:

- I think this relies on the quirk that `str` doesn't fulfill this protocol because its `__contains__` doesn't accept `object`. That's subtle enough that it's worth explicitly commenting on.
- Unlike `Sequence`, this doesn't provide helper methods like `.index` (https://github.com/python/typeshed/blob/bbd9dd1c4f596f564542d48bb05b2cc2e2a7a28d/stdlib/typing.pyi#L516). We should either add those or document the difference. Probably we should add them, as someone might ask for them later and adding methods to a protocol technically breaks compatibility.